modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvRow.java

```diff
@@ -155,7 +156,7 @@ public String field(int index) {
   public String getField(String header) {
     Integer index = findIndex(header);
     if (index == null) {
-      throw new IllegalArgumentException("Header not found: '" + header + "'");
+      throw new ParseFailureException("Header not found: '{header}'", header);
```

Comment:
Just need to update the Javadoc exception for this method and `getField(String header, Function<String, T> postProcessor)`

Reply:
Ah good spot - done